### PR TITLE
feat: [CDS-46672]: added miigration and made new beans mandatory and …

### PR DESCRIPTION
**Pull request by naman-goenka**
…deprecated old beans

## Describe your changes
- [ ] added migration to new yaml for SNow connector
- [ ]  made new beans mandatory
- [ ] marked old beans as deprecated ( but still have handling for them) , these fields will not been shown in yaml UI now.

https://user-images.githubusercontent.com/97941791/205335393-ec3b06ef-74a0-4a75-bb8c-ca789c98150a.mov

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/40608)

### CODE REVIEW

**Comment by naman-goenka** [comment]
trigger feature-build

**Review by puthrayaharness — DISMISSED** [review]

File: 120-ng-manager/src/main/java/io/harness/ng/core/migration/background/PopulateYamlAuthFieldInNGServiceNowConnectorMigration.java [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    try {
+      log.info(DEBUG_LOG + "Starting migration to ServiceNowAuthenticationDTO in ServiceNow connector");
+      List<ServiceNowConnector> serviceNowConnectors = new ArrayList<>();
+      try {
+        Criteria serviceNowConnectorCriteria =
+            Criteria.where(ServiceNowConnector.ConnectorKeys.type).is(ConnectorType.SERVICENOW);
+        Query serviceNowConnectorQuery = new Query(serviceNowConnectorCriteria);
+        serviceNowConnectors = mongoTemplate.find(serviceNowConnectorQuery, ServiceNowConnector.class);
+        if (isNull(serviceNowConnectors)) {
+          log.info(String.format("%s no serviceNow connectors fetched", DEBUG_LOG));
```
Comment (shivamnegi94) [inline 120-ng-manager/src/main/java/io/harness/ng/core/migration/background/PopulateYamlAuthFieldInNGServiceNowConnectorMigration.java:51]: if there is no serviceNow connector then why we are moving to next step ?

Comment (naman-goenka) [reply 120-ng-manager/src/main/java/io/harness/ng/core/migration/background/PopulateYamlAuthFieldInNGServiceNowConnectorMigration.java:51]: correct, adding a return; statement.

**Review by shivamnegi94 — COMMENTED** [review]
> Reviewed 6 of 7 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40608)* status: 7 file(s) to review. 1. unresolved discussions. 0 of 2 approvals obtained.

**Review by shivamnegi94 — APPROVED** [review]

**Comment by naman-goenka** [comment]
trigger ti0, ti1, ti2, ti3, ti4, ft1, ft2, checkstylecodeformat

**Comment by naman-goenka** [comment]
trigger smartchecks

**Comment by naman-goenka** [comment]
trigger ti1

**Review by shivamnegi94 — APPROVED** [review]

**Review by shivamnegi94 — COMMENTED** [review]
> Reviewed 2 of 2 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40608)* status: 7 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by vivekkarnatiharness — APPROVED** [review]
> Reviewed 4 of 7 files at r1, 1 of 1 files at r2, 1 of 2 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40608)* status: 1 file(s) to review. 1. unresolved discussions.

**Review by hingerabhinav — APPROVED** [review]
> Reviewed 4 of 7 files at r1, 1 of 1 files at r2, 2 of 2 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40608)* status: 1. unresolved discussions.

**Review by vivekkarnatiharness — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40608)* status: 1. unresolved discussions.

**Review by shivamnegi94 — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40608)* status: :shipit: complete! ✓

**Pull request merged**